### PR TITLE
User id fetch error

### PR DESCRIPTION
**Pull request by robertrmartinez**
## Type of change
- [X] Bugfix

## Description of change
If a user id submodule did failed on it's ajax request it would hold up the auction until the auctionDelay was hit.

Instead, we should have the callbacks called without params on a failure.

The following user id modules had the issue:

```
unifiedIdSystem
id5IdSystem
parrableIdSystem
liveIntentIdSystem
criteoIdSystem
identityLinkIdSystem
```

`criteoIdSystem` said they would make the update themselves.

This PR only adds a test for `parrableIdSystem` and `liveIntentIdSystem` because they are the only adapters which actually have a test `_spec` file.

Maybe we should change that? I thought it was required for all modules to have 80% test coverage.

### CODE REVIEW

File: modules/liveIntentIdSystem.js [hunk 1]
```diff
@@ -143,17 +143,24 @@ export const liveIntentIdSubmodule = {
...
+            try {
+              responseObj = JSON.parse(response);
+            } catch (error) {
+              utils.logError(error);
+            }
           }
+          callback(responseObj);
+        },
+        error: error => {
+          utils.logError(`${MODULE_NAME}: ID fetch encountered an error: `, error);
```
Comment (smenzer) [inline modules/liveIntentIdSystem.js:159]: @robertrmartinez is there a reason that sometimes you used `${MODULE_NAME}` and other times you just hardcoded the name? Just curious...

**Comment by bretg** [comment]
Alerting other ID module owners to these changes: @pycnvr, @mamatic, @jankoulaga, @melgenek, @icflournoy, @eyas-ranjous, @jlukas79 (unifiedId)

**Comment by icflournoy** [comment]
Thanks for the ping @bretg. The changes to `parrableIdSystem` look fine. Thanks for the added test @robertrmartinez.

I have some local work that improves our test coverage that I'll be pushing in a few days and will make sure to incorporate the changes in this PR once merged.

File: modules/liveIntentIdSystem.js (hunk 1 above)
Comment (robertrmartinez) [reply modules/liveIntentIdSystem.js:159]: Hi @smenzer 

`MODULE_NAME` is a common var adapters / modules use as to not replicate the literal string of their module name. But some do not utilize it.

So I only used `MODULE_NAME` in the adapters which actually have it already set.

**Comment by robertrmartinez** [comment]
@icflournoy That's good news! Always love improving test coverage!

**Comment by mamatic** [comment]
Thanks for the info @bretg. Yeah, it's looks good. Changes on identityLinkIdSystem.js is ok.

File: modules/liveIntentIdSystem.js (hunk 1 above)
Comment (smenzer) [reply modules/liveIntentIdSystem.js:159]: ah ok, good to know, thanks!

**Comment by mamatic** [comment]
Hello everyone, @bretg @bmwcmw @jlukas79 , just a double check, is this pull request going to be merged any time soon? We need to change few things on identityLinkIdSystem.

**Comment by bretg** [comment]
@robertrmartinez - please fix the conflict and merge.

@mamatic - I'm making the assumption that you want to make the identityLinkIdSystem changes after this one's merged?

**Comment by mamatic** [comment]
Yes, that is correct. After this pull request is merged we will make a new one. Thanks @bretg .

**Comment by mamatic** [comment]
Hi guys, just wanted to check what is the status of this PR? @bretg , @robertrmartinez

**Comment by robertrmartinez** [comment]
@mamatic I have just merged this, wanted to give plenty of time for the submodule maintainers to verify.

Thanks!

**Pull request merged**